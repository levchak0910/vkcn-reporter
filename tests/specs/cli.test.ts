import { execSync } from "node:child_process"

import { beforeAll, describe, expect, it } from "vitest"

const executeFailingCommand = (command: string): string => {
  let output: string | Error = new Error("should throw")
  try {
    execSync(command)
  } catch (error) {
    output = error.stdout.toString()
  }

  expect(output).toBeTypeOf("string")
  return output as string
}

describe("cli", () => {
  beforeAll(() => {
    execSync("pnpm build")
  })

  it("test wrong glob", async () => {
    const stdout = execSync("node dist/cli.js wrong-path")

    expect(stdout.toString()).toContain("No duplicate class selectors found")
    expect(stdout.toString()).toContain("No potentially leaking class selectors found")
  })

  it("test single glob", async () => {
    const output = executeFailingCommand("node dist/cli.js tests/mock/*.{css,scss,vue}")

    expect(output).toMatchSnapshot()
  })

  it("test multiple globs", async () => {
    const output = executeFailingCommand("node dist/cli.js 'tests/mock/*.css tests/mock/*.scss tests/mock/*.vue'")

    expect(output).toMatchSnapshot()
  })

  it("test glob with single ignore glob", async () => {
    const output = executeFailingCommand("node dist/cli.js tests/mock/*.{css,scss,vue} -i tests/mock/**/*.scss")

    expect(output).toMatchSnapshot()
  })

  it("test glob with multiple ignore glob", async () => {
    const output = executeFailingCommand(
      "node dist/cli.js tests/mock/*.{css,scss,vue} -i 'tests/mock/**/*.scss tests/mock/**/*.css'",
    )

    expect(output).toMatchSnapshot()
  })
})
