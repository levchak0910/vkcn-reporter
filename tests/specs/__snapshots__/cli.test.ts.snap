// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cli > test glob with multiple ignore glob 1`] = `
"‚ùå Found 2 duplicated class selectors

Class selector ".main-component--scss" defined in 2 files
  - tests/mock/main-component.vue
  - tests/mock/duplicate-component.vue

Class selector ".main-component--css" defined in 2 files
  - tests/mock/main-component.vue
  - tests/mock/duplicate-component.vue

üëç No potentially leaking class selectors found
"
`;

exports[`cli > test glob with single ignore glob 1`] = `
"‚ùå Found 2 duplicated class selectors

Class selector ".main-component--scss" defined in 2 files
  - tests/mock/main-component.vue
  - tests/mock/duplicate-component.vue

Class selector ".main-component--css" defined in 3 files
  - tests/mock/main-component.vue
  - tests/mock/main-component--css.css
  - tests/mock/duplicate-component.vue

‚ùå Found 2 potentially leaking class selectors

Class selector ".css-mod" can potentially leak in 2 files
  - tests/mock/main-component.vue
  - tests/mock/css-mod.css

Class selector ".css-mod-double" can potentially leak in 2 files
  - tests/mock/main-component.vue
  - tests/mock/css-mod-double.css

"
`;

exports[`cli > test multiple globs 1`] = `
"‚ùå Found 2 duplicated class selectors

Class selector ".main-component--css" defined in 3 files
  - tests/mock/main-component--css.css
  - tests/mock/main-component.vue
  - tests/mock/duplicate-component.vue

Class selector ".main-component--scss" defined in 3 files
  - tests/mock/main-component--scss.scss
  - tests/mock/main-component.vue
  - tests/mock/duplicate-component.vue

‚ùå Found 6 potentially leaking class selectors

Class selector ".scss-mod" can potentially leak in 3 files
  - tests/mock/main-component.vue
  - tests/mock/scss-mod.scss
  - tests/mock/scss-mod-parent.scss

Class selector ".scss-mod-double" can potentially leak in 2 files
  - tests/mock/main-component.vue
  - tests/mock/scss-mod-double.scss

Class selector ".scss-mod-nested" can potentially leak in 2 files
  - tests/mock/main-component.vue
  - tests/mock/scss-mod-nested.scss

Class selector ".scss-mod-nested-double" can potentially leak in 2 files
  - tests/mock/main-component.vue
  - tests/mock/scss-mod-nested-double.scss

Class selector ".css-mod" can potentially leak in 2 files
  - tests/mock/main-component.vue
  - tests/mock/css-mod.css

Class selector ".css-mod-double" can potentially leak in 2 files
  - tests/mock/main-component.vue
  - tests/mock/css-mod-double.css

"
`;

exports[`cli > test single glob 1`] = `
"‚ùå Found 2 duplicated class selectors

Class selector ".main-component--scss" defined in 3 files
  - tests/mock/main-component.vue
  - tests/mock/main-component--scss.scss
  - tests/mock/duplicate-component.vue

Class selector ".main-component--css" defined in 3 files
  - tests/mock/main-component.vue
  - tests/mock/main-component--css.css
  - tests/mock/duplicate-component.vue

‚ùå Found 6 potentially leaking class selectors

Class selector ".scss-mod" can potentially leak in 3 files
  - tests/mock/main-component.vue
  - tests/mock/scss-mod.scss
  - tests/mock/scss-mod-parent.scss

Class selector ".scss-mod-double" can potentially leak in 2 files
  - tests/mock/main-component.vue
  - tests/mock/scss-mod-double.scss

Class selector ".scss-mod-nested" can potentially leak in 2 files
  - tests/mock/main-component.vue
  - tests/mock/scss-mod-nested.scss

Class selector ".scss-mod-nested-double" can potentially leak in 2 files
  - tests/mock/main-component.vue
  - tests/mock/scss-mod-nested-double.scss

Class selector ".css-mod" can potentially leak in 2 files
  - tests/mock/main-component.vue
  - tests/mock/css-mod.css

Class selector ".css-mod-double" can potentially leak in 2 files
  - tests/mock/main-component.vue
  - tests/mock/css-mod-double.css

"
`;
